src/tiled/tileselectionitem.cpp

```diff
@@ -74,6 +76,12 @@ void TileSelectionItem::selectionChanged(const QRegion &newSelection,
     update(mMapDocument->renderer()->boundingRect(changedArea));
 }
 
+void TileSelectionItem::layerChanged(int index)
+{
+    const Layer *layer = mMapDocument->map()->layerAt(index);
```

Comment:
To avoid synchronizing the position to the wrong layer (mostly when using undo), we should add this check here:
```c++
if (layer == mMapDocument->currentLayer())
```